src/components/utils/src/threads/posix_thread.cc

```diff
@@ -279,13 +278,14 @@ void Thread::join() {
   LOGGER_AUTO_TRACE(logger_);
   DCHECK_OR_RETURN_VOID(!IsCurrentThread());
 
-  stop();
-
   sync_primitives::AutoLock auto_lock(state_lock_);
+
   run_cond_.NotifyOne();
-  if (isThreadRunning_) {
+  if (thread_state_ == RUNNING) {
```

Comment:
@anosach-luxoft , checked with this condition on timer too. On first timer test we have deadlock in join() method